xbmc/xbmc.cpp

```diff
@@ -53,7 +53,7 @@ extern "C" int XBMC_Run(bool renderGUI)
   g_RBP.LogFirmwareVerison();
 #endif
 
-  if (renderGUI && !g_application.CreateGUI())
+  if (renderGUI && !g_application.IsHeadless() && !g_application.CreateGUI())
```

Comment:
renderGUI comes from the method, i.e. its exposed to shared kodi library.
However perhaps they shouldn't be different and setting --headless should
just call with renderGUI=false and do a setheadleas(renderGUI).